raphtory-graphql/src/data.rs

```diff
@@ -67,6 +72,24 @@ impl Data {
         }
     }
 
+    pub fn new_graph(&self, path: &Path, graph_type: GqlGraphType) -> Result<(), GraphError> {
+        let full_path = self.construct_graph_full_path(path)?;
+        if full_path.exists() {
+            return Err(GraphError::GraphNameAlreadyExists(path.to_path_buf()).into());
+        }
+        create_dirs_if_not_present(&full_path)?;
+        let mut cache = File::create_new(full_path)?;
```

Comment:
this can still fail with the mysterious IO error here if someone manages to create the file between the previous check and this one. Probably better to only use this to check and map the error for a better message.